iOS/MSMapsModules/MSMapsModules/MSMapGeoJsonParser/MSMapGeoJsonParser.m

```diff
@@ -228,13 +296,47 @@ + (MSGeoposition * _Nullable)parseGeoposition:(NSArray *)coordinates
             @"com.microsoft.modules.MSMapsModules.MSMapGeoJsonParseError"
                    code:-500
                userInfo:@{
-                 NSLocalizedDescriptionKey : @"coordinates array must contain "
-                                             @"at least latitude and longitude."
+                 NSLocalizedDescriptionKey :
+                     [@"coordinates array must contain "
+                      @"at least latitude and longitude. Instead saw: "
+                         stringByAppendingString:
+                             [coordinates componentsJoinedByString:@", "]]
                }];
     return nil;
   }
 }
 
++ (MSGeopath *)parsePath:(NSArray *)pathArray error:(NSError **)error {
+  NSArray<MSGeoposition *> *path = [[NSArray alloc] init];
+  MSMapAltitudeReferenceSystem ref = MSMapAltitudeReferenceSystemSurface;
+  for (id obj in pathArray) {
+    if (![obj isKindOfClass:[NSArray class]]) {
+      *error = [NSError
+          errorWithDomain:
+              @"com.microsoft.modules.MSMapsModules.MSMapGeoJsonParseError"
+                     code:-500
+                 userInfo:@{
+                   NSLocalizedDescriptionKey :
+                       [@"coordinates of LineString must contain an array of "
+                        @"positions. Instead saw: "
+                           stringByAppendingFormat:@"%@", (NSString *)obj]
+                 }];
```

Comment:
Consider adding convenience function which takes argument(s) for the error message and creates the message for you. You have a lot of lines of identical code in each place where you create an error.